### PR TITLE
Fix maintenance pagination reset effect

### LINKED ISSUE #248: Tech debt: resolve React Doctor state reset error in MaintenancePageClient
## Context

React Doctor full-repo scan after merging PR #247 reports one remaining error outside the CalendarWidget scope.

## Finding

- Rule: `react-doctor/no-derived-state-effect`
- Severity: error
- File: `src/app/(app)/maintenance/_components/MaintenancePageClient.tsx:132`
- Current pattern:

```tsx
React.useEffect(() => {
  setCurrentPage(1)
}, [selectedFacilityId, debouncedPlanSearch])
```

React Doctor message:

> State reset in useEffect — use a key prop to reset component state when props change

## Reproduce

Run a true full scan, not diff-only:

```bash
printf '{"diff": false}' > react-doctor.config.json\nnode scripts/npm-run.js npx react-doctor@latest . --verbose --yes --project nextn --offline\ngit restore react-doctor.config.json\n```\n\nExpected current result includes:\n\n```text\n✗ State reset in useEffect — use a key prop to reset component state when props change\n  src/app/(app)/maintenance/_components/MaintenancePageClient.tsx: 132\n```\n\n## Acceptance Criteria\n\n- Remove or restructure the `setCurrentPage(1)` reset effect so React Doctor no longer reports `react-doctor/no-derived-state-effect` for `MaintenancePageClient`.\n- Preserve pagination behavior: changing facility filter or debounced plan search still returns the maintenance plan list to page 1.\n- Add or update focused coverage for that pagination reset behavior if practical.\n- Run TypeScript/React verification gates for the changed files.\n\n## Notes\n\nThis is not part of CalendarWidget issue #240; it was discovered during the post-merge full React Doctor scan.

### LINKED ISSUE #249: Tech debt: resolve React Doctor event-handler warning in MaintenancePageClient
## Context

While verifying Issue #248 on branch `issue-248-maintenance-pagination-reset`, React Doctor no longer reports the original `react-doctor/no-derived-state-effect` error, but it still reports a separate warning in the same component.

## Finding

- Rule/message: `useEffect simulating an event handler — move logic to an actual event handler instead`
- Severity: warning
- File: `src/app/(app)/maintenance/_components/MaintenancePageClient.tsx:108`
- Current pattern:

```tsx
React.useEffect(() => {
  if (facilitiesError) {
    console.error("[Maintenance] Failed to fetch facilities:", facilitiesError)
  }
}, [facilitiesError])
```

## Reproduce

Run:

```bash
node scripts/npm-run.js npx react-doctor@latest . --verbose -y --project nextn --offline --diff main
```

Current output includes the warning at `MaintenancePageClient.tsx:108`.

## Acceptance Criteria

- React Doctor diff scan no longer reports the `useEffect simulating an event handler` warning for `MaintenancePageClient`.
- Preserve facility query error observability or replace it with the project-preferred error reporting pattern.
- Keep this separate from Issue #248, which only fixes the state-reset error.
- Run TypeScript/React verification gates for changed files.

## Related

- Follow-up from #248 verification.

### PR DESCRIPTION
**Pull request by thienchi2109**
## Summary
- Extract maintenance plan list controls into a local hook so filter/search/page-size handlers reset pagination directly.
- Remove the React Doctor `no-derived-state-effect` reset effect from `MaintenancePageClient`.
- Remove the redundant facilities error logging effect that React Doctor flagged in follow-up #249; RPC failures remain logged by `callRpc`.
- Add focused `user-event` regression tests for facility/search pagination reset behavior.

Closes #248
Closes #249

## Test Plan
- [x] `node scripts/npm-run.js run verify:no-explicit-any`
- [x] `node scripts/npm-run.js run typecheck`
- [x] `node scripts/npm-run.js run test:run -- 'src/app/(app)/maintenance/__tests__/MaintenanceKpi.test.tsx'`
- [x] `node scripts/npm-run.js npx react-doctor@latest . --verbose -y --project nextn --offline --diff main`
- [x] `node scripts/npm-run.js npx react-doctor@latest . --verbose --no --project nextn --offline` (before #249 follow-up; full scan had no errors but retained unrelated warnings/non-fatal dead-code exception)

## Notes
- React Doctor diff scan now reports `No issues found!` for changed files.
- Full React Doctor scan previously showed no errors after #248; it still has existing full-repo warnings outside this PR and a non-fatal dead-code detector exception: `TypeError: issues.files is not iterable`.

## Summary by CodeRabbit

* **Refactor**
  * Centralized maintenance plan list controls for consistent search, facility, pagination and mobile filter behavior across layouts.

* **Bug Fixes**
  * Pagination now reliably resets to page 1 when changing facility, page size, or performing a search; mobile filter apply/clear and sheet open/close stay synchronized to avoid stale queries.

* **Tests**
  * Added integration tests and helpers to verify pagination reset semantics and filter/search interactions.

### CODE REVIEW

**Comment by thienchi2109** [comment]
Review follow-up pushed in `7a417d8`.\n\nHandled:\n- Verified the inline debounced-search/page navigation finding with a focused failing test, then fixed it in `useMaintenancePlanListControls`.\n- Added explicit return types for the new test helpers and exported hook return shape.\n\nIntentionally not changed:\n- The alias-import nitpick. `MaintenancePageClient` and nearby maintenance module files already use co-located `../_hooks` imports consistently; changing only this file would be style churn rather than a behavior or safety fix.\n\nVerification after the follow-up:\n- `node scripts/npm-run.js run verify:no-explicit-any`\n- `node scripts/npm-run.js run typecheck`\n- `node scripts/npm-run.js run test:run -- 'src/app/(app)/maintenance/__tests__/MaintenanceKpi.test.tsx'` -> 9/9\n- `node scripts/npm-run.js npx react-doctor@latest . --verbose -y --project nextn --offline --diff main` -> No issues found

File: src/app/(app)/maintenance/_hooks/use-maintenance-plan-list-controls.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const [planSearchTerm, setPlanSearchTerm] = React.useState("")
+  const [selectedFacilityId, setSelectedFacilityId] = React.useState<number | null>(null)
+  const [currentPage, setCurrentPage] = React.useState(1)
+  const [pageSize, setPageSize] = React.useState(50)
+  const [isMobileFilterSheetOpen, setIsMobileFilterSheetOpen] = React.useState(false)
+  const [pendingFacilityFilter, setPendingFacilityFilter] = React.useState<number | null>(null)
+
+  const handlePlanSearchChange = React.useCallback((value: string) => {
+    setPlanSearchTerm(value)
+    setCurrentPage(1)
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/(app)/maintenance/_hooks/use-maintenance-plan-list-controls.ts:47]: Verified this against a focused regression test and it was a real edge case. Added coverage for search input changing, user navigating before debounce settles, then debounced search applying; updated `useMaintenancePlanListControls` to return `currentPage: 1` in the same render where debounced search changes while also syncing state for the next render. Pushed in `7a417d8`.
